e2e/_suites/kubernetes-autodiscover/features/beats.feature

```diff
@@ -0,0 +1,52 @@
+@beats
+Feature: Beats
+  Use Kubernetes autodiscover features in Beats to monitor pods
+
+Scenario: Pod is started
+  Given a cluster is available
+    And configuration for "filebeat" has "hints enabled"
+    And "filebeat" is running
```

Comment:
wdyt about having one single step for that? Having multiple ANDs is a smell that we could be "scripting". Ideally zero or one AND is acceptable, although if properly discussed it can be accepted

```suggestion
    And "filebeat" is running with a configuration that has "hints enabled"
```

Reply:
I am on the fence here, but I will let you advise me :slightly_smiling_face: 

There are two main reasons why I did it this way:
* I expect different ways of deploying resources, this is why there are different "is running" and "is deployed" steps, the first one would be for pods that I would expect to be running before continuing with the next step, and the second one for resources in general (including pods that are never going to reach the running state, as failing or short living ones).
* I expect the configurations for some of these resources to have any number of "options" to chose different configuration blocks.

The alternatives I saw for this would be to have more specific steps or more specific templates.

More specific steps would be similar to what you propose, there would be steps such as:
* `"a pod" is running`
* `"a pod" is deployed`
* `"a pod" is running with a configuration that has "hints for metrics"`
* `"a pod" is deployed with a configuration that has "hints for metrics"`

We would need to implement each one of the different combinations, and we would be limited to zero or one options. If we want to include templates with more than one variation we would need to implement some additional step, or make the templates aware of more complex options. But I am not sure if we need that, so maybe this is not such a problem.

Maybe another option is to make the desired state a parameter, so we only have two steps that would allow to use one or none options:

* `"a pod" is "running"`
* `"a pod" is "deployed"`
* `"a pod" is "running" with a configuration that has "hints for metrics"`
* `"a pod" is "deployed" with a configuration that has "hints for metrics"`

One extra point in favour of that is that we don't need to keep state as I am keeping now in the `configurations` map. But this state is also used in the deletion steps, we would need to ensure that for the same pod we are using the same options in the run/deploy and delete steps.

With more specific templates we would have:
* `"a pod" is running`
* `"a pod" is deployed`
* `"a pod with hints for metrics" is running`
* `"a pod with hints for metrics" is deployed`

That would require to have many more templates.

Reply:
Ok, now I get your point. What do you think about having more scenarios, with just the required steps: one `When` and one `Then`, and probably one `And` with each of them?

Reply:
I have implemented your proposal at the end, reusing most of the underlying code and removing this `configurations` state. It actually simplifies things. Let me know what you think: 8ac0978bfb776c3f71f82af83978c6d6f0f79c37